### PR TITLE
ARM image tests

### PR DESCRIPTION
**Pull request by hopkiw**
* Add `-machine_type` support to CIT
* Add image tests to `debian-11-arm64` image build
* Use metadata lookup for interface lookups in tests rather than predictive naming

### CODE REVIEW

File: imagetest/cmd/manager/main.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ var (
 	parallelCount = flag.Int("parallel_count", 5, "TestParallelCount")
 	filter        = flag.String("filter", "", "only run tests matching filter")
 	exclude       = flag.String("exclude", "", "skip tests matching filter")
+	machineType   = flag.String("machine_type", "", "machine type")
```
Comment (zoran15) [inline imagetest/cmd/manager/main.go:38]: I guess machine type for VMs running the test? Extra description would be useful.

File: imagetest/testworkflow.go [hunk 2]
```diff
@@ -272,8 +272,19 @@ func finalizeWorkflows(ctx context.Context, tests []*TestWorkflow, zone, gcsPref
...
-		twf.wf.Sources["testpackage"] = fmt.Sprintf("/%s.test", twf.Name)
+		arch := "amd64"
+		if machineType != "" {
+			createVMsStep := twf.wf.Steps[createVMsStepName]
+			for _, vm := range createVMsStep.CreateInstances.Instances {
+				vm.MachineType = machineType
+			}
+			if strings.HasPrefix(machineType, "t2a") {
+				arch = "arm64"
+			}
```
Comment (zoran15) [inline imagetest/testworkflow.go:283]: Is this logic going to hold forever? I'm not very familiar with how machine types map to the architecture.

**Review by zoran15 — APPROVED** [review]
Only reviewed the Go part of the code. A couple of minor comments.

File: imagetest/testworkflow.go (hunk 2 above)
Comment (hopkiw) [reply imagetest/testworkflow.go:283]: t2a will always be ARM, but there may be new ARM types in the future. the Right Thing To Do™ here is to use compute API and read arch from machine type, but that's not yet available in prod API, and would also be the first direct use of compute API in this code.

Comment (zoran15) [reply imagetest/testworkflow.go:283]: SGTM. For tracking, I would personally create a bug to update this code once the API is available.

**Review by zoran15 — APPROVED** [review]

**Pull request merged**